src/engine/test/source/hlp/hlp_test.cpp

```diff
@@ -95,16 +97,19 @@ TEST(hlpTests_logQL, optional_Or)
     static const char *eventNone = "Mon Jan 2 15:04:05 MST 2006";
 
     auto parseOp = getParserOp(logQl);
-    auto resultJSON = parseOp(eventJSON);
+    ParseResult resultJSON;
+    bool ret = parseOp(eventJSON, resultJSON);
     ASSERT_EQ(true, static_cast<bool>(parseOp));
     ASSERT_EQ("{\"String\":\"SomeValue\"}", resultJSON["_field"]);
 
-    auto resultURL = parseOp(eventURL);
+    ParseResult resultURL;
+    ret = parseOp(eventURL, resultURL);
     std::string url = "https://user:password@wazuh.com:8080/"
                       "path?query=%22a%20query%20with%20a%20space%22#fragment";
     ASSERT_EQ(url, resultURL["url.original"]);
 
-    auto resultEmpty = parseOp(eventNone);
+    ParseResult resultEmpty;
+    ret = parseOp(eventNone, resultEmpty);
```

Comment:
We should add on every test the check that **bool ret** is **true** for success scenarios and **false** for error scenarios.

Reply:
yeah, i think i left a TODO for that